File: lib/src/main/java/uk/iatom/rfm69/registers/values/BroadcastAddress.java
package uk.iatom.rfm69.registers.values; import uk.iatom.rfm69.registers.IRegisterValue;

public enum BroadcastAddress implements IRegisterValue {

    // RegBroadcastAdrs),
    VALUE(0x00);

    private byte val;

    BroadcastAddress(int val) {
        this.val = (byte) val;
    }

    @Override public byte val() {
        return val;
    }
}

File: lib/src/main/java/uk/iatom/rfm69/registers/values/BroadcastAddress.kt
package uk.iatom.rfm69.registers.values

import uk.iatom.rfm69.registers.IRegisterValue

enum class BroadcastAddress(`val`: Int): IRegisterValue { // RegBroadcastAdrs),
VALUE(0x00);

    private val `val`: Byte

    init {
        this.`val` = `val`.toByte()
    }

    override fun `val`(): Byte {
        return `val`
    }
}
